src/pybind/mgr/dashboard/tests/test_prometheus.py

```diff
@@ -79,6 +79,11 @@ def test_post_on_receiver(self):
         self.assertEqual(notification['name'], 'foo')
         self.assertTrue(len(notification['notified']) > 20)
 
+        # test with empty body
+        self._post('/api/prometheus_receiver', {})
```

Comment:
Are you sure this is an empty request? Actually it has a body with an empty object, which is not the same as an empty body... To trigger this issue this should send a request without any data in the body at all.

Reply:
Actually the fact that we are now triggering an exception and this test passes without catching that exception means that the test is not covering the code path you fixed.